src/topology/mod.rs

```diff
@@ -75,7 +75,11 @@ pub async fn start_validated(
     config: Config,
     diff: ConfigDiff,
     mut pieces: Pieces,
-) -> Option<(RunningTopology, mpsc::UnboundedReceiver<()>)> {
+) -> Option<(
+    RunningTopology,
+    mpsc::UnboundedSender<()>,
+    mpsc::UnboundedReceiver<()>,
+)> {
```

Comment:
Given that most of the sites ignore the sender and receiver, it might have been simpler to turn the second return value into a tuple:
```suggestion
) -> Option<(
    RunningTopology,
    (mpsc::UnboundedSender<()>, mpsc::UnboundedReceiver<()>),
)> {
```

Reply:
Good point, I'll change it.